### ISSUE TITLE
FK as PK with IdentifiedReference

### ISSUE THREAD

**Issue by ml1nk** [mikro-orm/mikro-orm#1111]
**Describe the bug**
Using a wrapped OneToOne with primary: true does not work.

~~~javascript
export class CoreNodeEntity {
  @PrimaryKey()
  id!: integer
}
~~~
~~~javascript
export class CoreRoleEntity {
  [PrimaryKeyType]: number
  [PrimaryKeyProp]: 'node'
  @OneToOne({ primary: true, onDelete: 'cascade', onUpdateIntegrity: 'cascade' })
  node!: IdentifiedReference<CoreNodeEntity>

  @OneToMany(() => CorePrivilegeEntity, _ => _.role, { eager: true, orphanRemoval: true })
  privileges = new Collection<CorePrivilegeEntity>(this)
}
~~~
~~~javascript
export class CorePrivilegeEntity {
  @PrimaryKey()
  id!: integer

  @ManyToOne({ onDelete: 'cascade', onUpdateIntegrity: 'cascade' })
  role!: CoreRoleEntity

  @Property()
  type!: number
}
~~~

Add CorePrivilegeEntity to CorePrivilegeEntity.

**Stack trace**
~~~javascript
DriverException: insert into "core_privilege" ("role_node_id", "type") values ('[object Object]', 10), ('[object Object]', 20), ('[object Object]', 22), ('[object Object]', 23), ('[object Object]', 300), ('[object Object]', 301), ('[object Object]', 302), ('[object Object]', 303) returning "id" - invalid input syntax for type integer: "[object Object]"
    at PostgreSqlExceptionConverter.convertException (/home/mlink/workspace/pitapi/node_modules/@mikro-orm/core/platforms/ExceptionConverter.js:8:16)
    at PostgreSqlExceptionConverter.convertException (/home/mlink/workspace/pitapi/node_modules/@mikro-orm/postgresql/PostgreSqlExceptionConverter.js:40:22)
    at PostgreSqlDriver.convertException (/home/mlink/workspace/pitapi/node_modules/@mikro-orm/core/drivers/DatabaseDriver.js:171:54)
    at /home/mlink/workspace/pitapi/node_modules/@mikro-orm/core/drivers/DatabaseDriver.js:175:24
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
    at async PostgreSqlDriver.nativeInsertMany (/home/mlink/workspace/pitapi/node_modules/@mikro-orm/knex/AbstractSqlDriver.js:191:19)
    at async ChangeSetPersister.persistNewEntitiesBatch (/home/mlink/workspace/pitapi/node_modules/@mikro-orm/core/unit-of-work/ChangeSetPersister.js:79:21)
    at async ChangeSetPersister.persistNewEntities (/home/mlink/workspace/pitapi/node_modules/@mikro-orm/core/unit-of-work/ChangeSetPersister.js:72:13)
    at async UnitOfWork.commitCreateChangeSets (/home/mlink/workspace/pitapi/node_modules/@mikro-orm/core/unit-of-work/UnitOfWork.js:472:9)
    at async UnitOfWork.persistToDatabase (/home/mlink/workspace/pitapi/node_modules/@mikro-orm/core/unit-of-work/UnitOfWork.js:434:13)

previous error: insert into "core_privilege" ("role_node_id", "type") values ('[object Object]', 10), ('[object Object]', 20), ('[object Object]', 22), ('[object Object]', 23), ('[object Object]', 300), ('[object Object]', 301), ('[object Object]', 302), ('[object Object]', 303) returning "id" - invalid input syntax for type integer: "[object Object]"
    at Parser.parseErrorMessage (/home/mlink/workspace/pitapi/node_modules/pg-protocol/src/parser.ts:357:11)
    at Parser.handlePacket (/home/mlink/workspace/pitapi/node_modules/pg-protocol/src/parser.ts:186:21)
    at Parser.parse (/home/mlink/workspace/pitapi/node_modules/pg-protocol/src/parser.ts:101:30)
    at Socket.<anonymous> (/home/mlink/workspace/pitapi/node_modules/pg-protocol/src/index.ts:7:48)
    at Socket.emit (node:events:329:20)
    at Socket.EventEmitter.emit (node:domain:467:12)
    at addChunk (node:internal/streams/readable:304:12)
    at readableAddChunk (node:internal/streams/readable:279:9)
    at Socket.Readable.push (node:internal/streams/readable:218:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:192:23) {
  length: 120,
  severity: 'ERROR',
  code: '22P02',
  detail: undefined,
  hint: undefined,
  position: '63',
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'numutils.c',
  line: '259',
  routine: 'pg_strtoint32'
}
~~~

**Additional context**
It does not seem to extract the id from IdentifiedReference but instead try to use it as sql param directly.
![image](https://user-images.githubusercontent.com/4943440/99972340-b0d89a00-2d9e-11eb-914a-fa600c2d51c7.png)

**Versions**

| Dependency | Version |
| node | 15.2.1 |
| typescript | 4.1.2 |
| mikro-orm | 4.3.1 |  ts-morph
| pg | 8.5.1 |

**Comment by B4nan** [comment]
How does your code look like? Entity definitions are not enough to reproduce anything more than a schema generator issues.

Ideally provide executable repro, that way I can start working on the fix right ahead. Also, as I mentioned earlier, your entity definitions are not even valid typescript so how can I really know what your code looks like? :] 

Regarding the screenshot, this is too late stage to debug it, the transformation to PK happens earlier - inside `ChangeSetComputer`.

Btw, having all entities prefixed with `Core` is really giving me hard times when trying to understand your code :/

**Comment by ml1nk** [comment]
Sorry for my bad bug report, i will try to always add a test case for you in the future.

It seems single inserts are working but multiple inserts are broken in this case:
https://github.com/mikro-orm/mikro-orm/pull/1116

**Comment by B4nan** [comment]
Fixed in `4.3.2-dev.6`
